linkerd/proxy/api-resolve/src/metadata.rs

```diff
@@ -48,7 +44,6 @@ impl Metadata {
             protocol_hint: ProtocolHint::Unknown,
             identity: None,
             weight: 10_000,
-            authority_override: None,
```

Comment:
Is removing `authority_override` code related to this change?

Reply:
No, thanks. This is a merge-o